File: src/main/java/kitchenpos/products/tobe/ProductRepository.java
package kitchenpos.products.tobe;

import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Repository
public interface ProductRepository {

    Product save(Product product);

    Optional<Product> findById(UUID id);

    List<Product> findAll();

    List<Product> findAllByIdIn(List<UUID> ids);
}

File: src/main/kotlin/kitchenpos/product/tobe/domain/ProductRepository.kt
package kitchenpos.product.tobe.domain

import java.util.*
import org.springframework.context.annotation.Primary
import org.springframework.stereotype.Repository

@Repository("tobeProductRepository")
@Primary
interface ProductRepository {
    fun save(product: Product): Product

    fun findById(id: UUID): Optional<Product>

    fun findAll(): List<Product>

    fun findAllByIdIn(ids: List<UUID>): List<Product>
}
